Consider using an interface library for all third party dependencies to silence warnings when compiling them

### Is your feature request related to a problem?

Currently we still receive a lot of warnings from third party libraries like these:

> /home/johannes/Dokumente/Inexor/vulkan-renderer/build/_deps/imgui-src/imgui.h:2751:43: warning: first argument in call to 'memset' is a pointer to non-trivially copyable type 'ImDrawListSplitter' [-Wnontrivial-memcall]
 2751 |     inline ImDrawListSplitter()  { memset(this, 0, sizeof(*this)); }
      |                                           ^
/home/johannes/Dokumente/Inexor/vulkan-renderer/build/_deps/imgui-src/imgui.h:2751:43: note: explicitly cast the pointer to silence this warning
 2751 |     inline ImDrawListSplitter()  { memset(this, 0, sizeof(*this)); }
      |                                           ^
      |                                           (void*)
/home/johannes/Dokumente/Inexor/vulkan-renderer/build/_deps/imgui-src/imgui.h:2822:60: warning: first argument in call to 'memset' is a pointer to non-trivially copyable type 'ImDrawList' [-Wnontrivial-memcall]
 2822 |     ImDrawList(ImDrawListSharedData* shared_data) { memset(this, 0, sizeof(*this)); _Data = shared_data; }
      |                                                            ^
/home/johannes/Dokumente/Inexor/vulkan-renderer/build/_deps/imgui-src/imgui.h:2822:60: note: explicitly cast the pointer to silence this warning
 2822 |     ImDrawList(ImDrawListSharedData* shared_data) { memset(this, 0, sizeof(*this)); _Data = shared_data; }
      |                                                            ^
      |                                                            (void*)
2 warnings generated.



### Description

These warnings are not helpful at all. It's not originating from our own source code and we will not fix them.
It turns out silencing these warnings is a bit harder than I thought. One possible solution would be to precompile every third party library into a core library, while silencing all warnings related to these third party libs. Our code can then use this core library, and we can focus on warnings from our own code.

### Alternatives

If we don't do something about this, the warning spam during compilation will continue.

### Affected Code

The third party libraries.

### Operating System

All

### Additional Context

None

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -129,7 +129,6 @@ jobs:
             -DVULKAN_SDK=$VULKAN_SDK \
             -DINEXOR_BUILD_BENCHMARKS=ON \
             -DINEXOR_BUILD_TESTS=ON \
-            -DCMAKE_CXX_FLAGS="-Wall -Wextra" \
             -GNinja \
             ${{ matrix.config.cmake_configure_options }}
 

--- a/benchmarks/CMakeLists.txt
+++ b/benchmarks/CMakeLists.txt
@@ -27,13 +27,26 @@ set_target_properties(
 
 target_link_libraries(
     inexor-vulkan-renderer-benchmarks
+# enable warnings for our own code
+target_compile_options(inexor-vulkan-renderer-benchmarks PRIVATE
+    $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-Wall -Wextra -Wpedantic>
+    $<$<CXX_COMPILER_ID:MSVC>:/W4>
+)
 
-    PRIVATE
+target_link_libraries(inexor-vulkan-renderer-benchmarks PRIVATE
     inexor-vulkan-renderer-core-lib
 
     PUBLIC
     benchmark::benchmark
 )
+
+# suppress all compiler warnings for third-party dependencies
+foreach(_dep gtest benchmark)
+    target_compile_options(${_dep} PRIVATE
+        $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-w>
+        $<$<CXX_COMPILER_ID:MSVC>:/W0>
+    )
+endforeach()
 
 if(MSVC)
     # Group dependencies in Visual Studio

--- a/benchmarks/CMakeLists.txt
+++ b/benchmarks/CMakeLists.txt
@@ -17,16 +17,12 @@ else()
     )
 endif()
 
-set_target_properties(
-    inexor-vulkan-renderer-benchmarks PROPERTIES
-
+set_target_properties(inexor-vulkan-renderer-benchmarks PROPERTIES
     CXX_EXTENSIONS OFF
     CXX_STANDARD 20
     CXX_STANDARD_REQUIRED ON
 )
 
-target_link_libraries(
-    inexor-vulkan-renderer-benchmarks
 # enable warnings for our own code
 target_compile_options(inexor-vulkan-renderer-benchmarks PRIVATE
     $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-Wall -Wextra -Wpedantic>
@@ -60,9 +56,8 @@ if(MSVC)
         gtest_main
     )
     # Move all dependencies into a subfolder
-    set_target_properties(${THIRD_PARTY_TARGETS}
-        PROPERTIES
-            EXCLUDE_FROM_ALL TRUE
-            FOLDER "Dependencies"
+    set_target_properties(${THIRD_PARTY_TARGETS} PROPERTIES
+        EXCLUDE_FROM_ALL TRUE
+        FOLDER "Dependencies"
     )
 endif()

--- a/dependencies.cmake
+++ b/dependencies.cmake
@@ -8,6 +8,7 @@ set(FETCHCONTENT_QUIET OFF)
 # Benchmarking library for inexor-vulkan-renderer-benchmarks
 set(BENCHMARK_ENABLE_TESTING OFF CACHE BOOL "Disable Google Benchmark tests" FORCE)
 FetchContent_Declare(benchmark
+    SYSTEM
     GIT_REPOSITORY https://github.com/google/benchmark
     GIT_TAG v1.8.5
     GIT_SHALLOW ON
@@ -16,6 +17,7 @@ FetchContent_Declare(benchmark
     
 # Command line argument parsing library
 FetchContent_Declare(CLI11
+    SYSTEM
     GIT_REPOSITORY https://github.com/CLIUtils/CLI11.git
     GIT_TAG v2.6.1
     GIT_SHALLOW ON
@@ -24,6 +26,7 @@ FetchContent_Declare(CLI11
 
 # Formatting library
 FetchContent_Declare(fmt
+    SYSTEM
     GIT_REPOSITORY https://github.com/fmtlib/fmt.git
     GIT_TAG 11.0.1
     GIT_SHALLOW ON
@@ -35,6 +38,7 @@ set(GLFW_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
 set(GLFW_BUILD_DOCS OFF CACHE BOOL "" FORCE)
 set(GLFW_BUILD_TESTS OFF CACHE BOOL "" FORCE)
 FetchContent_Declare(glfw
+    SYSTEM
     GIT_REPOSITORY https://github.com/glfw/glfw.git
     GIT_TAG 3.4
     GIT_SHALLOW ON
@@ -43,6 +47,7 @@ FetchContent_Declare(glfw
 
 # Mathematics library for computer graphics
 FetchContent_Declare(glm
+    SYSTEM
     GIT_REPOSITORY https://github.com/g-truc/glm.git
     GIT_TAG 1.0.1
     GIT_SHALLOW ON
@@ -50,6 +55,7 @@ FetchContent_Declare(glm
 
 # Unit testing library for inexor-vulkan-renderer-tests
 FetchContent_Declare(gtest
+    SYSTEM
     GIT_REPOSITORY https://github.com/google/googletest
     GIT_TAG v1.15.0
     GIT_SHALLOW ON
@@ -58,6 +64,7 @@ FetchContent_Declare(gtest
 
 # Graphical user interface rendering library
 FetchContent_Declare(imgui
+    SYSTEM
     GIT_REPOSITORY https://github.com/ocornut/imgui.git
     GIT_TAG v1.90.9
     GIT_SHALLOW ON
@@ -68,6 +75,7 @@ FetchContent_Declare(imgui
 set(SPDLOG_USE_STD_FORMAT ON CACHE BOOL "" FORCE)
 set(SPDLOG_FMT_EXTERNAL OFF CACHE BOOL "" FORCE)
 FetchContent_Declare(spdlog
+    SYSTEM
     GIT_REPOSITORY https://github.com/gabime/spdlog.git
     GIT_TAG v1.15.3
     GIT_SHALLOW ON
@@ -76,6 +84,7 @@ FetchContent_Declare(spdlog
 
 # Required for loading textures in various formats
 FetchContent_Declare(stb
+    SYSTEM
     GIT_REPOSITORY https://github.com/nothings/stb.git
     GIT_TAG f7f20f39fe4f206c6f19e26ebfef7b261ee59ee4
     # Do not enable shallow download on stb because it will cause a CMake error!
@@ -83,6 +92,7 @@ FetchContent_Declare(stb
 
 # Loader for models in glTF2 format
 FetchContent_Declare(tinygltf
+    SYSTEM
     GIT_REPOSITORY https://github.com/syoyo/tinygltf.git
     GIT_TAG v2.9.2
     GIT_SHALLOW ON
@@ -91,6 +101,7 @@ FetchContent_Declare(tinygltf
 
 # Library for TOML configuration files
 FetchContent_Declare(tomlplusplus
+    SYSTEM
     GIT_REPOSITORY https://github.com/marzer/tomlplusplus.git
     GIT_TAG v3.4.0
     GIT_PROGRESS ON
@@ -98,6 +109,7 @@ FetchContent_Declare(tomlplusplus
 
 # Vulkan Memory Allocator library
 FetchContent_Declare(vma
+    SYSTEM
     GIT_REPOSITORY https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator.git
     GIT_TAG v3.1.0
     GIT_SHALLOW ON
@@ -106,6 +118,7 @@ FetchContent_Declare(vma
 
 # Meta-loader for the Vulkan API that dynamically loads function pointers at runtime, avoiding the need for static linking
 FetchContent_Declare(volk
+    SYSTEM
     GIT_REPOSITORY https://github.com/zeux/volk
     GIT_TAG 1.3.270
     GIT_SHALLOW ON
@@ -116,6 +129,7 @@ FetchContent_Declare(volk
 
 # The Vulkan API headers
 FetchContent_Declare(Vulkan
+    SYSTEM
     GIT_REPOSITORY https://github.com/KhronosGroup/Vulkan-Headers
     GIT_TAG v1.3.283
     GIT_SHALLOW ON

--- a/dependencies.cmake
+++ b/dependencies.cmake
@@ -7,118 +7,117 @@ set(FETCHCONTENT_QUIET OFF)
 
 # Benchmarking library for inexor-vulkan-renderer-benchmarks
 set(BENCHMARK_ENABLE_TESTING OFF CACHE BOOL "Disable Google Benchmark tests" FORCE)
-FetchContent_Declare(benchmark
-    SYSTEM
+FetchContent_Declare(benchmark SYSTEM
     GIT_REPOSITORY https://github.com/google/benchmark
     GIT_TAG v1.8.5
     GIT_SHALLOW ON
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 1.8.5)
+    FIND_PACKAGE_ARGS 1.8.5
+)
     
 # Command line argument parsing library
-FetchContent_Declare(CLI11
-    SYSTEM
+FetchContent_Declare(CLI11 SYSTEM
     GIT_REPOSITORY https://github.com/CLIUtils/CLI11.git
     GIT_TAG v2.6.1
     GIT_SHALLOW ON
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 2.6.1)
+    FIND_PACKAGE_ARGS 2.6.1
+)
 
 # Formatting library
-FetchContent_Declare(fmt
-    SYSTEM
+FetchContent_Declare(fmt SYSTEM
     GIT_REPOSITORY https://github.com/fmtlib/fmt.git
     GIT_TAG 11.0.1
     GIT_SHALLOW ON
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 11.0.1)
+    FIND_PACKAGE_ARGS 11.0.1
+)
 
 # Window management for Linux and Microsoft Windows
 set(GLFW_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
 set(GLFW_BUILD_DOCS OFF CACHE BOOL "" FORCE)
 set(GLFW_BUILD_TESTS OFF CACHE BOOL "" FORCE)
-FetchContent_Declare(glfw
-    SYSTEM
+FetchContent_Declare(glfw SYSTEM
     GIT_REPOSITORY https://github.com/glfw/glfw.git
     GIT_TAG 3.4
     GIT_SHALLOW ON
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 3.4)
+    FIND_PACKAGE_ARGS 3.4
+)
 
 # Mathematics library for computer graphics
-FetchContent_Declare(glm
-    SYSTEM
+FetchContent_Declare(glm SYSTEM
     GIT_REPOSITORY https://github.com/g-truc/glm.git
     GIT_TAG 1.0.1
     GIT_SHALLOW ON
-    GIT_PROGRESS ON)
+    GIT_PROGRESS ON
+)
 
 # Unit testing library for inexor-vulkan-renderer-tests
-FetchContent_Declare(gtest
-    SYSTEM
+FetchContent_Declare(gtest SYSTEM
     GIT_REPOSITORY https://github.com/google/googletest
     GIT_TAG v1.15.0
     GIT_SHALLOW ON
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 1.15.0)
+    FIND_PACKAGE_ARGS 1.15.0
+)
 
 # Graphical user interface rendering library
-FetchContent_Declare(imgui
-    SYSTEM
+FetchContent_Declare(imgui SYSTEM
     GIT_REPOSITORY https://github.com/ocornut/imgui.git
     GIT_TAG v1.90.9
     GIT_SHALLOW ON
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 1.90.9)
+    FIND_PACKAGE_ARGS 1.90.9
+)
 
 # Console logging and logfile library
 set(SPDLOG_USE_STD_FORMAT ON CACHE BOOL "" FORCE)
 set(SPDLOG_FMT_EXTERNAL OFF CACHE BOOL "" FORCE)
-FetchContent_Declare(spdlog
-    SYSTEM
+FetchContent_Declare(spdlog SYSTEM
     GIT_REPOSITORY https://github.com/gabime/spdlog.git
     GIT_TAG v1.15.3
     GIT_SHALLOW ON
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 1.15.3)
+    FIND_PACKAGE_ARGS 1.15.3
+)
 
 # Required for loading textures in various formats
-FetchContent_Declare(stb
-    SYSTEM
+FetchContent_Declare(stb SYSTEM
     GIT_REPOSITORY https://github.com/nothings/stb.git
     GIT_TAG f7f20f39fe4f206c6f19e26ebfef7b261ee59ee4
     # Do not enable shallow download on stb because it will cause a CMake error!
-    GIT_PROGRESS ON)
+    GIT_PROGRESS ON
+)
 
 # Loader for models in glTF2 format
-FetchContent_Declare(tinygltf
-    SYSTEM
+FetchContent_Declare(tinygltf SYSTEM
     GIT_REPOSITORY https://github.com/syoyo/tinygltf.git
     GIT_TAG v2.9.2
     GIT_SHALLOW ON
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 2.9.2)
+    FIND_PACKAGE_ARGS 2.9.2
+)
 
 # Library for TOML configuration files
-FetchContent_Declare(tomlplusplus
-    SYSTEM
+FetchContent_Declare(tomlplusplus SYSTEM
     GIT_REPOSITORY https://github.com/marzer/tomlplusplus.git
     GIT_TAG v3.4.0
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 3.4.0)
+    FIND_PACKAGE_ARGS 3.4.0
+)
 
 # Vulkan Memory Allocator library
-FetchContent_Declare(vma
-    SYSTEM
+FetchContent_Declare(vma SYSTEM
     GIT_REPOSITORY https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator.git
     GIT_TAG v3.1.0
     GIT_SHALLOW ON
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 3.1.0)
+    FIND_PACKAGE_ARGS 3.1.0
+)
 
 # Meta-loader for the Vulkan API that dynamically loads function pointers at runtime, avoiding the need for static linking
-FetchContent_Declare(volk
-    SYSTEM
+FetchContent_Declare(volk SYSTEM
     GIT_REPOSITORY https://github.com/zeux/volk
     GIT_TAG 1.3.270
     GIT_SHALLOW ON
@@ -128,10 +127,10 @@ FetchContent_Declare(volk
 )
 
 # The Vulkan API headers
-FetchContent_Declare(Vulkan
-    SYSTEM
+FetchContent_Declare(Vulkan SYSTEM
     GIT_REPOSITORY https://github.com/KhronosGroup/Vulkan-Headers
     GIT_TAG v1.3.283
     GIT_SHALLOW ON
     GIT_PROGRESS ON
-    FIND_PACKAGE_ARGS 1.3.283)
+    FIND_PACKAGE_ARGS 1.3.283
+)

--- a/documentation/CMakeLists.txt
+++ b/documentation/CMakeLists.txt
@@ -38,8 +38,7 @@ message(STATUS "Found Sphinx: ${SPHINX_EXECUTABLE}")
 
 # Sphinx cache with pickled ReST documents
 set(SPHINX_CACHE_DIR "${CMAKE_CURRENT_BINARY_DIR}/_doctrees")
-add_custom_target(
-    inexor-vulkan-renderer-documentation ALL
+add_custom_target(inexor-vulkan-renderer-documentation ALL
     COMMAND ${CMAKE_COMMAND} -E remove_directory "${CMAKE_CURRENT_BINARY_DIR}/_doctrees"
     COMMAND ${CMAKE_COMMAND} -E remove_directory "${CMAKE_CURRENT_BINARY_DIR}/html"
     COMMAND ${CMAKE_COMMAND} -E remove_directory "${CMAKE_CURRENT_SOURCE_DIR}/source/auto-generated"
@@ -48,8 +47,7 @@ add_custom_target(
     COMMAND ${SPHINX_EXECUTABLE} -j auto -T -E -a -b html -d "${SPHINX_CACHE_DIR}" "${CMAKE_CURRENT_SOURCE_DIR}/source" "${CMAKE_CURRENT_BINARY_DIR}/html"
     COMMENT "Building HTML documentation with Sphinx"
 )
-add_custom_target(
-    inexor-vulkan-renderer-documentation-linkcheck
+add_custom_target(inexor-vulkan-renderer-documentation-linkcheck
     COMMAND ${SPHINX_EXECUTABLE} -j auto -T -b linkcheck -d "${SPHINX_CACHE_DIR}" "${CMAKE_CURRENT_SOURCE_DIR}/source" "${CMAKE_CURRENT_BINARY_DIR}/html"
     COMMENT "Building linkcheck documentation with Sphinx"
 )

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -77,18 +77,13 @@ add_library(inexor-vulkan-renderer-core-lib ${INEXOR_SOURCE_FILES})
 
 add_dependencies(inexor-vulkan-renderer-core-lib inexor-shaders)
 
-set_target_properties(
-    inexor-vulkan-renderer-core-lib PROPERTIES
-
+set_target_properties(inexor-vulkan-renderer-core-lib PROPERTIES
     CXX_EXTENSIONS OFF
     CXX_STANDARD 20
     CXX_STANDARD_REQUIRED ON
 )
 
-target_compile_definitions(
-    inexor-vulkan-renderer-core-lib
-
-    PUBLIC
+target_compile_definitions(inexor-vulkan-renderer-core-lib PUBLIC
     GLFW_INCLUDE_VULKAN
     GLM_ENABLE_EXPERIMENTAL
     GLM_FORCE_DEPTH_ZERO_TO_ONE
@@ -114,22 +109,19 @@ if(MSVC)
     target_compile_options(inexor-vulkan-renderer-core-lib PRIVATE "/Zc:__cplusplus")
 endif()
 
-# Replace variables in header file with CMake values
+# replace variables in header file with CMake values
 configure_file(
     ${PROJECT_SOURCE_DIR}/include/inexor/vulkan-renderer/meta/meta.hpp.in
     ${CMAKE_CURRENT_BINARY_DIR}/include/inexor/vulkan-renderer/meta/meta.hpp
 )
 
-target_include_directories(
-    inexor-vulkan-renderer-core-lib
-
-    PUBLIC
+target_include_directories(inexor-vulkan-renderer-core-lib PUBLIC
     ${PROJECT_SOURCE_DIR}/include
-    # Include configured header file which contains engine metadata
+    # include configured header file which contains engine metadata
     ${CMAKE_CURRENT_BINARY_DIR}/include/
 )
 
-# Declare use of dependencies
+# declare use of dependencies
 FetchContent_MakeAvailable(CLI11)
 FetchContent_MakeAvailable(fmt)
 FetchContent_MakeAvailable(spdlog)
@@ -146,11 +138,11 @@ add_library(imgui
     ${imgui_SOURCE_DIR}/imgui.cpp
     ${imgui_SOURCE_DIR}/imgui_draw.cpp
     ${imgui_SOURCE_DIR}/imgui_tables.cpp
-    ${imgui_SOURCE_DIR}/imgui_widgets.cpp)
+    ${imgui_SOURCE_DIR}/imgui_widgets.cpp
+)
 target_include_directories(imgui SYSTEM PUBLIC ${imgui_SOURCE_DIR})
 
-target_link_libraries(inexor-vulkan-renderer-core-lib
-    PUBLIC
+target_link_libraries(inexor-vulkan-renderer-core-lib PUBLIC
     CLI11::CLI11
     fmt::fmt
     glfw
@@ -161,7 +153,8 @@ target_link_libraries(inexor-vulkan-renderer-core-lib
     tomlplusplus::tomlplusplus
     volk::volk
     Vulkan::Headers
-    VulkanMemoryAllocator)
+    VulkanMemoryAllocator
+)
 
 # suppress all compiler warnings for third-party dependencies
 foreach(_dep glfw glm imgui tinygltf)
@@ -172,7 +165,7 @@ foreach(_dep glfw glm imgui tinygltf)
 endforeach()
 
 if(MSVC)
-    # Group dependencies in Visual Studio
+    # group dependencies in Visual Studio
     set(THIRD_PARTY_TARGETS
         CLI11
         fmt
@@ -183,15 +176,14 @@ if(MSVC)
         tinygltf
         volk
         VulkanMemoryAllocator
-        # Also those extra ones
+        # also those extra ones
         loader_example
         uninstall
         update_mappings
     )
-    # Move all dependencies into a subfolder
-    set_target_properties(${THIRD_PARTY_TARGETS}
-        PROPERTIES
-            EXCLUDE_FROM_ALL TRUE
-            FOLDER "Dependencies"
+    # move all dependencies into a subfolder
+    set_target_properties(${THIRD_PARTY_TARGETS} PROPERTIES
+        EXCLUDE_FROM_ALL TRUE
+        FOLDER "Dependencies"
     )
 endif()

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -96,6 +96,12 @@ target_compile_definitions(
     VK_NO_PROTOTYPES # Required by volk metaloader
 )
 
+# enable warnings for our own code
+target_compile_options(inexor-vulkan-renderer-core-lib PRIVATE
+    $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-Wall -Wextra -Wpedantic>
+    $<$<CXX_COMPILER_ID:MSVC>:/W4>
+)
+
 # enable multi processor building if VS project
 if(${CMAKE_GENERATOR} MATCHES "Visual Studio.*")
     target_compile_options(inexor-vulkan-renderer-core-lib PRIVATE "/MP")
@@ -141,7 +147,7 @@ add_library(imgui
     ${imgui_SOURCE_DIR}/imgui_draw.cpp
     ${imgui_SOURCE_DIR}/imgui_tables.cpp
     ${imgui_SOURCE_DIR}/imgui_widgets.cpp)
-target_include_directories(imgui PUBLIC ${imgui_SOURCE_DIR})
+target_include_directories(imgui SYSTEM PUBLIC ${imgui_SOURCE_DIR})
 
 target_link_libraries(inexor-vulkan-renderer-core-lib
     PUBLIC
@@ -156,6 +162,14 @@ target_link_libraries(inexor-vulkan-renderer-core-lib
     volk::volk
     Vulkan::Headers
     VulkanMemoryAllocator)
+
+# suppress all compiler warnings for third-party dependencies
+foreach(_dep glfw glm imgui tinygltf)
+    target_compile_options(${_dep} PRIVATE
+        $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-w>
+        $<$<CXX_COMPILER_ID:MSVC>:/W0>
+    )
+endforeach()
 
 if(MSVC)
     # Group dependencies in Visual Studio

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -35,10 +35,7 @@ target_compile_options(inexor-vulkan-renderer-tests PRIVATE
     $<$<CXX_COMPILER_ID:MSVC>:/W4>
 )
 
-target_link_libraries(
-    inexor-vulkan-renderer-tests
-
-    PRIVATE
+target_link_libraries(inexor-vulkan-renderer-tests PRIVATE
     inexor-vulkan-renderer-core-lib
 
     PUBLIC
@@ -62,9 +59,8 @@ if(MSVC)
         gtest_main
     )
     # Move all dependencies into a subfolder
-    set_target_properties(${THIRD_PARTY_TARGETS}
-        PROPERTIES
-            EXCLUDE_FROM_ALL TRUE
-            FOLDER "Dependencies"
+    set_target_properties(${THIRD_PARTY_TARGETS} PROPERTIES
+        EXCLUDE_FROM_ALL TRUE
+        FOLDER "Dependencies"
     )
 endif()

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -29,6 +29,12 @@ set_target_properties(
     CXX_STANDARD_REQUIRED ON
 )
 
+# enable warnings for our own code
+target_compile_options(inexor-vulkan-renderer-tests PRIVATE
+    $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-Wall -Wextra -Wpedantic>
+    $<$<CXX_COMPILER_ID:MSVC>:/W4>
+)
+
 target_link_libraries(
     inexor-vulkan-renderer-tests
 
@@ -38,6 +44,14 @@ target_link_libraries(
     PUBLIC
     GTest::gtest
 )
+
+# suppress all compiler warnings for third-party dependencies
+foreach(_dep gtest)
+    target_compile_options(${_dep} PRIVATE
+        $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-w>
+        $<$<CXX_COMPILER_ID:MSVC>:/W0>
+    )
+endforeach()
 
 if(MSVC)
     # Group dependencies in Visual Studio

